test/triangular.jl

```diff
@@ -0,0 +1,65 @@
+@testset "Triangular-matrix multiplication" begin
```

Comment:
Cool, I can definitely add an `srand`. Also, I'm happy to spend some time writing explicit test cases if you'd prefer that. Another way of reducing test time is to test only for a single value of `n`. When there are specific methods for each size they definitely need to be tested independently but with a general method I'm not sure how much you gain by testing multiple sizes.